Pytest fails

Hi, 
I am trying to package `icontract-hypothesis` for `nixpkgs`. When running `pytest` I am getting the following errors:
```
==================================== ERRORS ====================================
____________________________ ERROR at setup of test ____________________________
file /build/source/icontract_hypothesis/pyicontract_hypothesis/_test.py, line 304
  def test(general: _general.Params, command: Params, stdout: TextIO) -> List[str]:
E       fixture 'general' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, monkeypatch, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/build/source/icontract_hypothesis/pyicontract_hypothesis/_test.py:304
=================================== FAILURES ===================================
________________________ TestSequence.test_sequence_int ________________________

self = <tests.strategy_inference.test_strategy_inference.TestSequence testMethod=test_sequence_int>

    def test_sequence_int(self) -> None:
        # noinspection PyUnusedLocal
        def some_func(xs: Sequence[int]) -> None:
            pass

        strategy = icontract_hypothesis.infer_strategy(some_func)

        # This might seem very surprising, but it is indeed the desired behavior,
        # see: https://github.com/HypothesisWorks/hypothesis/issues/2950
>       self.assertEqual(
            "fixed_dictionaries({'xs': one_of(binary(), lists(integers()))})",
            str(strategy),
        )
E       AssertionError: "fixe[51 chars]rs()))})" != "fixe[51 chars]rs()), fixed_dictionaries({'xs': one_of(binary[147 chars]))})"
E       - fixed_dictionaries({'xs': one_of(binary(), lists(integers()))})
E       + fixed_dictionaries({'xs': one_of(binary(), lists(integers()), fixed_dictionaries({'xs': one_of(binary().filter(lambda xs: all(x > -(2 ** 63) for x in xs)), lists(integers()).filter(lambda xs: all(x > -(2 ** 63) for x in xs)))}).map(lambda d: C(**d)))})

tests/strategy_inference/test_strategy_inference.py:844: AssertionError
____________________ TestForwardDeclarations.test_function _____________________

self = <test_strategy_inference_on_forward_declarations.TestForwardDeclarations testMethod=test_function>

    def test_function(self) -> None:
        strategy = icontract_hypothesis.infer_strategy(do_something)

>       self.assertEqual(
            "fixed_dictionaries("
            "{'a': "
            "fixed_dictionaries({'x': integers()})"
            ".map(lambda d: A(**d))})",
            str(strategy),
        )
E       AssertionError: "fixe[16 chars]'a': fixed_dictionaries({'x': integers()}).map[16 chars]))})" != "fixe[16 chars]'a': builds(A)})"
E       - fixed_dictionaries({'a': fixed_dictionaries({'x': integers()}).map(lambda d: A(**d))})
E       + fixed_dictionaries({'a': builds(A)})

tests_3_7/strategy_inference/test_strategy_inference_on_forward_declarations.py:49: AssertionError
_________________ TestForwardDeclarations.test_instance_method _________________

self = <test_strategy_inference_on_forward_declarations.TestForwardDeclarations testMethod=test_instance_method>

    def test_instance_method(self) -> None:
        b = B(0)

        strategy = icontract_hypothesis.infer_strategy(b.do_something)

>       self.assertEqual(
            "fixed_dictionaries({'x1': integers()})",
            str(strategy),
        )
E       AssertionError: "fixed_dictionaries({'x1': integers()})" != "fixed_dictionaries({'self': just(<test_strategy_inference_on_f[66 chars]()})"
E       - fixed_dictionaries({'x1': integers()})
E       + fixed_dictionaries({'self': just(<test_strategy_inference_on_forward_declarations.B object at 0x7ffff59c5eb0>),
E        'x1': integers()})

tests_3_7/strategy_inference/test_strategy_inference_on_forward_declarations.py:64: AssertionError
_______________________ TestForwardDeclarations.test_new _______________________

self = <test_strategy_inference_on_forward_declarations.TestForwardDeclarations testMethod=test_new>

    def test_new(self) -> None:
        strategy = icontract_hypothesis.infer_strategy(some_func_on_c)

>       self.assertEqual(
            "fixed_dictionaries("
            "{'c': "
            "fixed_dictionaries("
            "{'xs': one_of(binary(), lists(integers()))"
            ".filter(lambda xs: all(x > -(2 ** 63) for x in xs))})"
            ".map(lambda d: C(**d))})",
            str(strategy),
        )
E       AssertionError: "fixe[57 chars]ary(), lists(integers())).filter(lambda xs: al[52 chars]))})" != "fixe[57 chars]ary().filter(lambda xs: all(x > -(2 ** 63) for[103 chars]))})"
E       Diff is 662 characters long. Set self.maxDiff to None to see it.

tests_3_7/strategy_inference/test_strategy_inference_on_forward_declarations.py:74: AssertionError
=============================== warnings summary ===============================
tests/pyicontract_hypothesis/test_test.py:208
  /build/source/tests/pyicontract_hypothesis/test_test.py:208: DeprecationWarning: invalid escape sequence \(
    out = re.sub("\(time delta [^)]*\)", "(time delta <erased>)", stdout.getvalue())

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/strategy_inference/test_strategy_inference.py::TestSequence::test_sequence_int
FAILED tests_3_7/strategy_inference/test_strategy_inference_on_forward_declarations.py::TestForwardDeclarations::test_function
FAILED tests_3_7/strategy_inference/test_strategy_inference_on_forward_declarations.py::TestForwardDeclarations::test_instance_method
FAILED tests_3_7/strategy_inference/test_strategy_inference_on_forward_declarations.py::TestForwardDeclarations::test_new
ERROR icontract_hypothesis/pyicontract_hypothesis/_test.py::test
============== 4 failed, 150 passed, 1 warning, 1 error in 13.73s ==============
```

Tested with `icontract` `2.6.1`, `hypothesis` `6.4.0` and python `3.7` `3.8` `3.9` `3.10`. The errors stay the same. 

Do you have any ideas?

--- a/tests_3_7/strategy_inference/test_strategy_inference_on_forward_declarations.py
+++ b/tests_3_7/strategy_inference/test_strategy_inference_on_forward_declarations.py
@@ -10,6 +10,7 @@
 import unittest
 from typing import Sequence
 
+import hypothesis
 import icontract
 
 import icontract_hypothesis
@@ -71,15 +72,27 @@ class TestForwardDeclarations(unittest.TestCase):
     def test_new(self) -> None:
         strategy = icontract_hypothesis.infer_strategy(some_func_on_c)
 
-        self.assertEqual(
-            "fixed_dictionaries("
-            "{'c': "
-            "fixed_dictionaries("
-            "{'xs': one_of(binary(), lists(integers()))"
-            ".filter(lambda xs: all(x > -(2 ** 63) for x in xs))})"
-            ".map(lambda d: C(**d))})",
-            str(strategy),
-        )
+        if hypothesis.__version_info__ <= (6, 14, 0):
+            self.assertEqual(
+                "fixed_dictionaries("
+                "{'c': "
+                "fixed_dictionaries("
+                "{'xs': one_of(binary(), lists(integers()))"
+                ".filter(lambda xs: all(x > -(2 ** 63) for x in xs))})"
+                ".map(lambda d: C(**d))})",
+                str(strategy),
+            )
+        else:
+            self.assertEqual(
+                "fixed_dictionaries("
+                "{'c': "
+                "fixed_dictionaries("
+                "{'xs': one_of("
+                "binary().filter(lambda xs: all(x > -(2 ** 63) for x in xs)), "
+                "lists(integers()).filter(lambda xs: all(x > -(2 ** 63) for x in xs)))})"
+                ".map(lambda d: C(**d))})",
+                str(strategy),
+            )
 
         icontract_hypothesis.test_with_inferred_strategy(do_something)
 

